examples/type_conversion.rs

```diff
@@ -0,0 +1,97 @@
+    // - `as` never panics and silently performs a saturating cast. It should
+    //   only be used for simple conversions.
+    //
+    // For a brush-up on casting between numeric types in Rust, refer to:
+    // https://doc.rust-lang.org/rust-by-example/types/cast.html
+
+    // Infallible, safe conversion with `From`
+    // The trait `std::convert::From` is only implemented for conversions that
+    // can be guaranteed to be safe at compile time. This is the safest
+    // approach.
+    let a_u8: Array<u8, _> = array![[1, 2, 3], [4, 5, 6]];
+    let a_f32 = a_u8.mapv(|element| f32::from(element));
+    assert_abs_diff_eq!(a_f32, array![[1.0, 2.0, 3.0], [4.0, 5.0, 6.0]]);
+
+    // Fallible, safe conversion with `TryFrom`
+    // `i8` numbers can be negative, in such a case, there is no perfect
+    // conversion to `u8` defined. In this example, all numbers are positive and
+    // in bounds and can be converted at runtime. But for unknown runtime input,
+    // this would panic with the message provided in `.expect()`. Note that you
+    // can also use `.unwrap()` to be more concise.
+    let a_i8: Array<i8, _> = array![120, 8, 0];
+    let a_u8 = a_i8.mapv(|element| u8::try_from(element).expect("Could not convert i8 to u8"));
+    assert_eq!(a_u8, array![120u8, 8u8, 0u8]);
+
+    // Simple upcasting with `as`
+    // Every `u8` fits perfectly into a `u32`, therefore we do not need to take
+    // special care and can use `as`.
+    let a_u8: Array<u8, _> = array![[1, 2, 3], [4, 5, 6]];
+    let a_u32 = a_u8.mapv(|element| element as u32);
+    assert_eq!(a_u32, array![[1u32, 2u32, 3u32], [4u32, 5u32, 6u32]]);
+
+    // Int to float casting with `as`
+    // Every `i32` number can be represented as `f32` (although maybe not
+    // exactly equal), thus we can use `as` without special care.
+    let a_i32: Array<i32, _> = array![[1, -2, 3], [-4, 5, -6]];
+    let a_f32 = a_i32.mapv(|element| element as f32);
+    assert_abs_diff_eq!(a_f32, array![[1.0, -2.0, 3.0], [-4.0, 5.0, -6.0]]);
+
+    // Saturating cast with `as`
+    // The `as` keyword performs a *saturating cast* since Rust 1.45. This means
+    // that numbers which do not fit into the target datatype will silently be
+    // clipped to the maximum/minimum numbers.
+    // Since this is not obvious, we discourage the intentional use of casting
+    // with `as` with silent saturation and recommend a custom logic instead
+    // which makes the intent clear.
+    let a_f32: Array<f32, _> = array![
+        256.0,         // saturated to 255
+        255.7,         // saturated to 255
+        255.1,         // saturated to 255
+        254.7,         // rounded down to 254 by cutting the decimal part
+        254.1,         // rounded down to 254 by cutting the decimal part
+        -1.0,          // saturated to 0 on the lower end
+        f32::INFINITY, // saturated to 255
+        f32::NAN,      // converted to zero
```

Comment:
Just to be sure @adamreichold  - this is not UB, right? Otherwise, I'd at least add a comment.

Reply:
Yes, this is well-defined, c.f. https://doc.rust-lang.org/reference/expressions/operator-expr.html#type-cast-expressions (Generally speaking, Rust code that does not use `unsafe` will not have undefined behaviour. That casts of floating point numbers used to have was a bug which was resolved by fixing the saturating semantics.)

Reply:
A great, thank you for the link!